### PR TITLE
Changelog for 1.4.0

### PR DESCRIPTION
**Pull request by sbogolepov**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+# v1.4.0 (Aug 2020)
+  * Objective-C/Swift interop:
+    * Reworked exception handling ([GH-4191](https://github.com/JetBrains/kotlin-native/pull/4191))
```
Comment (SvyatoslavScherbina) [inline CHANGELOG.md:3]: Aren't GH-* links recognized automatically?

Comment (SvyatoslavScherbina) [inline CHANGELOG.md:3]: https://github.com/JetBrains/kotlin-native/pull/3822 and https://github.com/JetBrains/kotlin-native/pull/3842 would probably be more relevant.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,15 @@
...
+    * Enabled support for Objective-C generics by default ([GH-3778](https://github.com/JetBrains/kotlin-native/pull/3778))
+    * Support for Kotlin’s suspending functions ([GH-3915](https://github.com/JetBrains/kotlin-native/pull/3915))
+    * .dSYMs for release binaries ([GH-4085](https://github.com/JetBrains/kotlin-native/pull/4085))
```
Comment (SvyatoslavScherbina) [inline CHANGELOG.md:6]: Not actually related to interop.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,15 @@
...
+  * Added native-specific frontend checkers (implemented in the main Kotlin repository: [GH-3293](https://github.com/JetBrains/kotlin/pull/3293), [GH-3091](https://github.com/JetBrains/kotlin/pull/3091), [GH-3172](https://github.com/JetBrains/kotlin/pull/3172))
+  * Reworked cinterop under the hood
```
Comment (SvyatoslavScherbina) [inline CHANGELOG.md:8]: More details are required.
For example, it is worth mentioning that it improves compilation time with cinterop involved.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,15 @@
...
+  * Experimental faster memory allocator (`-Xallocator=mimalloc`) ([Native-3704](https://github.com/JetBrains/kotlin-native/pull/3704))
```
Comment (SvyatoslavScherbina) [inline CHANGELOG.md:9]: `Native` -> `GH`.

File: CHANGELOG.md (hunk 1 above)
Comment (sbogolepov) [reply CHANGELOG.md:3]: It may be useful for non-GitHub readers.

File: CHANGELOG.md (hunk 3 above)
Comment (raniejade) [reply CHANGELOG.md]: Out of curiosity, is the improvement in OSX only or across the board?

Comment (SvyatoslavScherbina) [reply CHANGELOG.md]: It affects all supported targets.

**Pull request merged**